northstar/src/runtime/process/mod.rs

```diff
@@ -165,155 +158,157 @@ impl Launcher {
             seccomp,
         };
 
-        // Fork trampoline process
-        match unsafe { unistd::fork() } {
-            Ok(result) => match result {
-                unistd::ForkResult::Parent { child } => {
-                    let trampoline_pid = child;
-                    // Close writing ends of log pipes (if any)
-                    drop(io);
-                    // Close child console socket (if any)
-                    drop(console_fd);
-                    // Close child checkpoint pipes
-                    drop(checkpoint_init);
-
-                    // Receive the pid of the init process from the trampoline process
-                    debug!("Waiting for the pid of init of {}", container);
-                    let mut channel = channel.into_async_read();
-                    let pid = channel.recv::<i32>().await.expect("Failed to read pid") as Pid;
-                    debug!("Created {} with pid {}", container, pid);
-
-                    // We're done reading the pid. The next information transferred via the
-                    // channel is the exit status of the container process.
-
-                    // Reap the trampoline process which is (or will be) a zombie otherwise
-                    debug!("Waiting for trampoline process {} to exit", trampoline_pid);
-                    wait::waitpid(Some(trampoline_pid), None)
-                        .expect("Failed to wait for trampoline process");
-
-                    // Start a task that waits for the exit of the init process
-                    let exit_status_fut = self.container_exit_status(container, channel, pid, stop);
-
-                    Ok(Process {
-                        pid,
-                        checkpoint: Some(checkpoint_runtime),
-                        exit_status: Some(Box::new(exit_status_fut)),
-                    })
-                }
-                unistd::ForkResult::Child => {
-                    // Forget writing ends of io which are stdout, stderr. The `forget`
-                    // ensures that the file descriptors are not closed
-                    forget(io);
+        // Channel just for receiving the init pid from the intermediate process
+        let mut init_pid_channel = channel::Channel::<u32>::new();
```

Comment:
I added this second pipe for sending just the pid to avoid having the `InitPid` variant in the `InitMessage` enum. I think it is a bit more independent that way :man_shrugging:  but is not a strong opinion

Reply:
Let's use the same channel. It's not that complicated and we safe the creation of another pipe.

Reply:
sure :man_shrugging:

Reply:
Well, on a second though :sweat_smile: using the same channel for sending the init's pid will force using `&mut` for passing the channel to the init and that required a `'static` for the receiver as we move it into a thread. In this case I think we are better with the second pipe :smile: